File: src/main/java/kr/co/dealmungchi/hotdealbatch/domain/entity/Provider.java
package kr.co.dealmungchi.hotdealbatch.domain.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AccessLevel;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "providers")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@AllArgsConstructor(access = AccessLevel.PRIVATE)
@Builder
public class Provider {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "provider_type", nullable = false)
    @Enumerated(value = EnumType.STRING)
    private ProviderType providerType;
}

File: src/main/kotlin/kr/co/dealmungchi/hotdealbatch/domain/entity/Provider.kt
package kr.co.dealmungchi.hotdealbatch.domain.entity

import jakarta.persistence.*

/**
 * Entity representing a deal provider.
 */
@Entity
@Table(name = "providers")
class Provider(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "provider_type", nullable = false)
    @Enumerated(value = EnumType.STRING)
    val providerType: ProviderType
)